src/appMain/life_cycle_impl.cc

```diff
@@ -80,9 +79,10 @@ LifeCycleImpl::LifeCycleImpl(const profile::Profile& profile)
 
 bool LifeCycleImpl::StartComponents() {
   LOG4CXX_AUTO_TRACE(logger_);
-  DCHECK(!last_state_);
-  last_state_ = new resumption::LastStateImpl(profile_.app_storage_folder(),
-                                              profile_.app_info_storage());
+  DCHECK(!last_state_wrapper_);
+  last_state_wrapper_ = std::make_shared<resumption::LastStateWrapperImpl>(
```

Comment:
@AByzhynar  I do not agree with such approach. LastStateWrapperImpl should know about Last state interface. But I would not say that Wrapper should create LastStateImpl.
If LastStateImpl will be extended with additional dependencies in the future - it should not affect LastStateWrapper. Also testing of last state wrapper would be more complicated if it will create LastStateImpl inside. 

If you have any additional arguments why LastStateWrapper whould create LastStateImpl inside please provide.

Reply:
@alexkutsan Ok. No problem. But create it in a separate line instead of mixing ctor into 2 make_shared operations. Such an approach makes it impossible to debug

Reply:
@AByzhynar , fixed in 1a5c410